mycroft/skills/settings.py

```diff
@@ -93,18 +93,22 @@ def get_local_settings(skill_dir, skill_name) -> dict:
 def save_settings(skill_dir, skill_settings):
     """Save skill settings to file."""
     settings_path = Path(skill_dir).joinpath('settings.json')
-    if Path(skill_dir).exists():
-        with open(str(settings_path), 'w') as settings_file:
-            try:
-                json.dump(skill_settings, settings_file)
-            except Exception:
-                LOG.exception('error saving skill settings to '
-                              '{}'.format(settings_path))
-            else:
-                LOG.info('Skill settings successfully saved to '
-                         '{}' .format(settings_path))
-    else:
-        LOG.info('Skill folder no longer exists, can\'t save settings.')
+
+    #: Either the file already exists in /opt, or we are writing
+    #: to XDG_CONFIG_DIR and always have the permission to make
+    #: sure the file always exists
+    if not Path(settings_path).exists():
+        settings_path.touch(mode=0o644)
```

Comment:
```python
def save_settings(skill_dir, skill_settings):
    """Save skill settings to file."""
    try:
        settings_path = Path(skill_dir).joinpath('settings.json')
        #: Either the file already exists in /opt, or we are writing
        #: to XDG_CONFIG_DIR and always have the permission to make
        #: sure the file always exists
        if not Path(settings_path).exists():
            settings_path.touch(mode=0o644)
        with open(str(settings_path), 'w') as settings_file:
            json.dump(skill_settings, settings_file)
    except Exception:
        LOG.exception('error saving skill settings to '
                      '{}'.format(settings_path))
    else:
        LOG.info('Skill settings successfully saved to '
                 '{}' .format(settings_path))

Reply:
It also strikes me that removed skills will save their settings here on their final shutdown. Which may not be a bad thing...

Reply:
The directory will at that point in the function always exists. If we're writing to a XDG directory, the folder will be created by `BaseDirectory.save_config_path()` if it does not exist yet, so you never have to check if the folder exists. In your local test you're probably not using `BaseDirectory.save_config_path()`, in which case the folder might not exist and will then indeed error out.

As for your last edit, in case of not saving to the skill directory, it should still work fine no? With your proposed change we'll not save the settings on shutdown ever, even if we would've saved it to the XDG directory which would've been fine.

Reply:
With the last edit this should be fine